recipes/vk-bootstrap/all/conanfile.py

```diff
@@ -47,60 +56,69 @@ def configure(self):
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 14)
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
 
-        def lazy_lt_semver(v1, v2):
+        def loose_lt_semver(v1, v2):
             lv1 = [int(v) for v in v1.split(".")]
             lv2 = [int(v) for v in v2.split(".")]
             min_length = min(len(lv1), len(lv2))
             return lv1[:min_length] < lv2[:min_length]
 
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if not minimum_version:
-            self.output.warn("vk-bootstrap requires C++14. Your compiler is unknown. Assuming it supports C++14.")
-        elif lazy_lt_semver(str(self.settings.compiler.version), minimum_version):
-            raise ConanInvalidConfiguration("vk-bootstrap requires C++14, which your compiler does not support.")
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and loose_lt_semver(str(self.info.settings.compiler.version), minimum_version):
+            raise ConanInvalidConfiguration(
+                f"{self.name} {self.version} requires C++{self._min_cppstd}, which your compiler does not support.",
+            )
 
-        if self._is_msvc and self.options.shared:
+        if is_msvc(self) and self.info.options.shared:
             raise ConanInvalidConfiguration("vk-boostrap shared not supported with Visual Studio")
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["VK_BOOTSTRAP_TEST"] = False
+        vulkan_headers = self.dependencies["vulkan-headers"]
+        includedirs = ";".join(
+            [os.path.join(vulkan_headers.package_folder, includedir).replace("\\", "/")
+             for includedir in vulkan_headers.cpp_info.includedirs],
+        )
+        if Version(self.version) < "0.3.0":
+            tc.variables["Vulkan_INCLUDE_DIR"] = includedirs
+        else:
+            tc.variables["VK_BOOTSTRAP_VULKAN_HEADER_DIR"] = includedirs
+        if Version(self.version) >= "0.4.0":
+            tc.variables["VK_BOOTSTRAP_WERROR"] = False
+        tc.generate()
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+        # Honor VK_BOOTSTRAP_TEST disabled
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
+            "if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR VK_BOOTSTRAP_TEST)",
+            "if(VK_BOOTSTRAP_TEST)",
+        )
```

Comment:
I'll backport https://github.com/charles-lunarg/vk-bootstrap/commit/4ae9513ff9182b9c519504a73435ed575a821300 to patches instead.